### PR TITLE
Bugzilla search

### PR DESCRIPTION
**Pull request by cpeters**

### CODE REVIEW

File: lib/tool_belt/bugzilla.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      :f4 => "flagtypes.name",
+      :o4 => "equals",
+      :v4 => "qa_ack+",
+      :f5 => "flagtypes.name",
+      :o5 => "equals",
+      :v5 => "sat-6.1.z+",
+      :query_format => "advanced",
+      :status => status,
+      :include_fields => default_fields,
+      :product => "Red Hat Satellite 6"
```
Comment (ehelms) [inline lib/tool_belt/bugzilla.rb]: Can we make this configurable versus hard-coding it here? Given the rest is bugzilla generic?

File: lib/tool_belt/bugzilla.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      :v2 => "pm_ack+",
+      :f3 => "flagtypes.name",
+      :o3 => "equals",
+      :v3 => "devel_ack+",
+      :f4 => "flagtypes.name",
+      :o4 => "equals",
+      :v4 => "qa_ack+",
+      :f5 => "flagtypes.name",
+      :o5 => "equals",
+      :v5 => "sat-6.1.z+",
```
Comment (ehelms) [inline lib/tool_belt/bugzilla.rb]: Can we make this configurable versus hard-coding it here?

File: lib/tool_belt/config.rb [hunk 3]
```diff
@@ -8,18 +8,37 @@ class Config
 
     def initialize(config_file, version=nil)
       configs = YAML.load_file(config_file)
-      valid?(configs)
-      configs = add_release(configs, version)
+      validate_bugzilla?(configs)
+      configs = add_bugzilla_release(configs, version)
+      #validate_redmine?(configs)
+      #configs = add_release(configs, version)
```
Comment (ehelms) [inline lib/tool_belt/config.rb]: We'll need this back or a switch of some sort here

Comment (cpeters) [inline lib/tool_belt/config.rb]: Which lines? 13 and 14 that were commented out?

Comment (cpeters) [inline lib/tool_belt/config.rb]: @ehelms I think I understand what you are talking about; we will need a method param that indicates whether or not to include bugzilla or not. Is that was you are referring to?

Comment (ehelms) [inline lib/tool_belt/config.rb]: There would essentially be two types of configuration files now -- Redmine and bugzilla that are formatted slightly differently and we need to be able to discern which is which and validate it properly.

**Comment by cpeters** [comment]
Don't Merge; fixing a mistake

**Comment by cpeters** [comment]
Fixed; ready for review.

File: lib/tool_belt/config.rb [hunk 4]
```diff
@@ -6,34 +6,64 @@ class Config
...
+
+      if version && !releases.map(&:to_s).include?(version)
+        fail "Unknown version specified. Please ensure the releases field is configured { X.Y.Z }."
       end
 
-      fail "Configuration file is incomplete. Please ensure it includes #{required_fields - configs.keys}." if !valid || configs[:releases].keys.empty?
-      fail "Configuration file is incomplete. Please ensure releases contains at least one entry of the form {version: X.Y, redmine_release_id: N}." if configs[:releases].keys.empty?
-      valid
+      configs[:release] = version.nil? ? releases.first.to_s : version.to_s
+      configs[:namespace] = "#{configs[:project]}_#{configs[:release]}"
```
Comment (ehelms) [inline lib/tool_belt/config.rb]: Should project be a required parameter?

Comment (cpeters) [inline lib/tool_belt/config.rb]: Yeah. Good catch!

**Comment by ehelms** [comment]
Thanks @cpeters !

**Pull request merged**